### ISSUE TITLE
IAS Zone produce unusable sensor

### ISSUE THREAD

**Issue by nerdosity** [doudz/homeassistant-zigate#1]
Hi, you still don't have documentation for supporting new devices.
I binded a smoke sensor from Heiman/generic, this one:

http://www.heimantech.com/product/89.html
I bough this from Aliexpress, it doesn't have the Heiman logo on, but it's the same sold from at least:

-Heiman (http://www.heimantech.com/product/89.html)
-Trust (https://www.trust.com/it/product/71197-zigbee-smoke-detector-zsdr-850)

This is for you, I hope it can help:

![image](https://user-images.githubusercontent.com/26657281/42726276-10d8140e-8792-11e8-80d0-2955ba28d4b0.png)

![image](https://user-images.githubusercontent.com/26657281/42726300-72f0639e-8792-11e8-8489-59beedfe6fa1.png)

Text:

`zigate_3a38_zone_status
59 secondi fa
{'trouble': False, 'alarm2': False, 'ac_fault': False, 'battery_defect': False, 'tamper': False, 'supervision': False, 'low_battery': False, 'restore': True, 'alarm1': False, 'test_mode': False}
cluster
1280
alarm2
false
tamper
false
low battery
false
addr
3a38
alarm1
false
test mode
false
trouble
false
ac fault
false
endpoint
1
supervision
false
battery defect
false
restore
true
attribute
0`

And:

```
3a38 1 minuto fa
2018-07-14 18:18:44
missing
false
type
SMOK_V16
receiver on when idle
true
max tx
0
battery level
100
max buffer
64
addr
3a38
manufacturer
bbaa
ieee
005043c903351f6a
bit field
0100000000000010
last seen
2018-07-14 18:18:44
rssi percent
13
server mask
0
max rx
0
descriptor capability
00000011
mac capability
10000100
battery
-
rssi
33
```

from homeassistant.log:

`2018-07-14 18:23:44 DEBUG (Thread-88) [zigate] RESPONSE 0x8401 - IAS Zone Status Change : sequence:0, endpoint:1, cluster:1280, address_mode:2, addr:3a38, zone_status:0000000000100001, status:0, zone_id:0, delay:0, rssi:33
2018-07-14 18:23:44 DEBUG (Thread-88) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2018-07-14 18:23:44 DEBUG (Thread-88) [zigate] Dispatch ZIGATE_ATTRIBUTE_UPDATED
2018-07-14 18:23:44 DEBUG (Thread-88) [custom_components.zigate] Update attribute for device Device 3a38 SMOK_V16 {'cluster': 1280, 'name': 'zone_status', 'zone_status': '0000000000100001', 'value': {'trouble': False, 'alarm2': False, 'ac_fault': False, 'battery_defect': False, 'tamper': False, 'supervision': False, 'low_battery': False, 'restore': True, 'alarm1': True, 'test_mode': False}, 'addr': '3a38', 'data': '0000000000100001', 'zone_id': 0, 'endpoint': 1, 'attribute': 0}
2018-07-14 18:23:45 DEBUG (Thread-89) [zigate] RESPONSE 0x8401 - IAS Zone Status Change : sequence:0, endpoint:1, cluster:1280, address_mode:2, addr:3a38, zone_status:0000000000100000, status:0, zone_id:0, delay:0, rssi:18
2018-07-14 18:23:45 DEBUG (Thread-89) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2018-07-14 18:23:45 DEBUG (Thread-89) [zigate] Dispatch ZIGATE_ATTRIBUTE_UPDATED
2018-07-14 18:23:45 DEBUG (Thread-89) [custom_components.zigate] Update attribute for device Device 3a38 SMOK_V16 {'cluster': 1280, 'name': 'zone_status', 'zone_status': '0000000000100000', 'value': {'trouble': False, 'alarm2': False, 'ac_fault': False, 'battery_defect': False, 'tamper': False, 'supervision': False, 'low_battery': False, 'restore': True, 'alarm1': False, 'test_mode': False}, 'addr': '3a38', 'data': '0000000000100000', 'zone_id': 0, 'endpoint': 1, 'attribute': 0}`

**Comment by doudz** [comment]
Well, thanks for sharing.
By the way, yes there's no doc about supported device, mainly because any device is supposed to be supported. 

Your device seems to work, what would like to change ?

**Comment by nerdosity** [comment]
Hello doudz,

I can make use of it 😄 but as a UI / Web developer I just can't stand it to see it as a bunch of JSON in home assistant, and that's even more important for a end user that should probably see a properly formatted item.
I would gladly help to have a polished product, I will start browsing your code for helping more properly, I don't have experience with python (I use mostly PHP / JS / Java / ASP) but I think I can read it without much problem.
Now, for suggestion:

- it (the smoke heisan sensor) should use a "trouble" icon / device type, there is a "smoke" one (actually it's the same as a problem)
https://www.home-assistant.io/components/binary_sensor/
That's pretty much errorless, since device name is "SMOK_V16" 😄 
- as far as I can see, the only thing that make this sensor "smart" is that it update the siren ringing status (alarm1 attribute).
That's exactly the same if you push the test button or if you let the alarm start for proper smoke condition.
So, in the sensor status, it should just let you see the status (activated or not) and maybe the time since last activation. I hoped it could monitor smoke data in the air, but it's just an on/off device, not really smart.

I think that this binary device should be something like that:

![image](https://user-images.githubusercontent.com/26657281/42734175-004f3e10-8840-11e8-87ad-6b70d893fe5d.png)

At a glance, battery level and another attribute, like "Last activated" or "No alarm since".
That's valid for every other device automagically created from this plugin, no end user cares about endpoint, cluster, attribute, address.

![image](https://user-images.githubusercontent.com/26657281/42734211-574f1eba-8840-11e8-8473-718912e7cc5c.png)
A door sensor on home assistant should let just me know quickly the battery status and from how much time it's opened/closed, like the official aqara plugin does:

![image](https://user-images.githubusercontent.com/26657281/42734217-7b8eddf6-8840-11e8-917b-722cdff7bc0f.png)

Every other attribute can be stuffed just inside the device on dev-state page, and it's even present in the "zigate" panel after clicking on it.


```
2018-07-15 14:28:08 DEBUG (Thread-478) [zigate] RESPONSE 0x8401 - IAS Zone Status Change : sequence:0, endpoint:1, cluster:1280, address_mode:2, addr:3a38, zone_status:0000000000100001, status:0, zone_id:0, delay:0, rssi:63
2018-07-15 14:28:08 DEBUG (Thread-478) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2018-07-15 14:28:08 DEBUG (Thread-478) [zigate] Dispatch ZIGATE_ATTRIBUTE_UPDATED
2018-07-15 14:28:08 DEBUG (Thread-478) [custom_components.zigate] Update attribute for device Device 3a38 SMOK_V16 {'data': '0000000000100001', 'name': 'zone_status', 'value': {'test_mode': False, 'alarm2': False, 'battery_defect': False, 'low_battery': False, 'alarm1': True, 'restore': True, 'ac_fault': False, 'supervision': False, 'tamper': False, 'trouble': False}, 'attribute': 0, 'zone_id': 0, 'endpoint': 1, 'cluster': 1280, 'zone_status': '0000000000100001', 'addr': '3a38'}
2018-07-15 14:28:10 DEBUG (Thread-479) [zigate] RESPONSE 0x8401 - IAS Zone Status Change : sequence:0, endpoint:1, cluster:1280, address_mode:2, addr:3a38, zone_status:0000000000100000, status:0, zone_id:0, delay:0, rssi:63
2018-07-15 14:28:10 DEBUG (Thread-479) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2018-07-15 14:28:10 DEBUG (Thread-479) [zigate] Dispatch ZIGATE_ATTRIBUTE_UPDATED
2018-07-15 14:28:10 DEBUG (Thread-479) [custom_components.zigate] Update attribute for device Device 3a38 SMOK_V16 {'data': '0000000000100000', 'name': 'zone_status', 'value': {'test_mode': False, 'alarm2': False, 'battery_defect': False, 'low_battery': False, 'alarm1': False, 'restore': True, 'ac_fault': False, 'supervision': False, 'tamper': False, 'trouble': False}, 'attribute': 0, 'zone_id': 0, 'endpoint': 1, 'cluster': 1280, 'zone_status': '0000000000100000', 'addr': '3a38'}
2018-07-15 14:28:20 DEBUG (Thread-480) [zigate] RESPONSE 0x8401 - IAS Zone Status Change : sequence:0, endpoint:1, cluster:1280, address_mode:2, addr:3a38, zone_status:0000000000100001, status:0, zone_id:0, delay:0, rssi:69
2018-07-15 14:28:20 DEBUG (Thread-480) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2018-07-15 14:28:20 DEBUG (Thread-480) [zigate] Dispatch ZIGATE_ATTRIBUTE_UPDATED
2018-07-15 14:28:20 DEBUG (Thread-480) [custom_components.zigate] Update attribute for device Device 3a38 SMOK_V16 {'data': '0000000000100001', 'name': 'zone_status', 'value': {'test_mode': False, 'alarm2': False, 'battery_defect': False, 'low_battery': False, 'alarm1': True, 'restore': True, 'ac_fault': False, 'supervision': False, 'tamper': False, 'trouble': False}, 'attribute': 0, 'zone_id': 0, 'endpoint': 1, 'cluster': 1280, 'zone_status': '0000000000100001', 'addr': '3a38'}
2018-07-15 14:28:40 DEBUG (Thread-481) [zigate] RESPONSE 0x8401 - IAS Zone Status Change : sequence:0, endpoint:1, cluster:1280, address_mode:2, addr:3a38, zone_status:0000000000100001, status:0, zone_id:0, delay:0, rssi:63
2018-07-15 14:28:40 DEBUG (Thread-481) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2018-07-15 14:28:40 DEBUG (Thread-481) [zigate] Dispatch ZIGATE_ATTRIBUTE_UPDATED
2018-07-15 14:28:40 DEBUG (Thread-481) [custom_components.zigate] Update attribute for device Device 3a38 SMOK_V16 {'data': '0000000000100001', 'name': 'zone_status', 'value': {'test_mode': False, 'alarm2': False, 'battery_defect': False, 'low_battery': False, 'alarm1': True, 'restore': True, 'ac_fault': False, 'supervision': False, 'tamper': False, 'trouble': False}, 'attribute': 0, 'zone_id': 0, 'endpoint': 1, 'cluster': 1280, 'zone_status': '0000000000100001', 'addr': '3a38'}
2018-07-15 14:28:46 DEBUG (Thread-482) [zigate] RESPONSE 0x8401 - IAS Zone Status Change : sequence:0, endpoint:1, cluster:1280, address_mode:2, addr:3a38, zone_status:0000000000100000, status:0, zone_id:0, delay:0, rssi:66
2018-07-15 14:28:46 DEBUG (Thread-482) [zigate] Dispatch ZIGATE_RESPONSE_RECEIVED
2018-07-15 14:28:46 DEBUG (Thread-482) [zigate] Dispatch ZIGATE_ATTRIBUTE_UPDATED
2018-07-15 14:28:46 DEBUG (Thread-482) [custom_components.zigate] Update attribute for device Device 3a38 SMOK_V16 {'data': '0000000000100000', 'name': 'zone_status', 'value': {'test_mode': False, 'alarm2': False, 'battery_defect': False, 'low_battery': False, 'alarm1': False, 'restore': True, 'ac_fault': False, 'supervision': False, 'tamper': False, 'trouble': False}, 'attribute': 0, 'zone_id': 0, 'endpoint': 1, 'cluster': 1280, 'zone_status': '0000000000100000', 'addr': '3a38'}

```

That was me:
- pushing test button
- waiting for it to stop ringing
- puffing a sigarette on it
- continuing puffing on it
- waiting for it to stop ringing

Yeah, not really smart, but it's enought for a clean binary sensor.

**Comment by doudz** [comment]
You're right, stuff like endpoint, cluster, ... are just for dev, not useful.
About the battery level, the information is already in the zigate entity, so no need to replicate this information, but it could be done easily.

You're right, the current state of the smoke sensor is not user friendly, but since the zone status contains many boolean attribute, the only thing I could do is create as many binary_sensor as attribute (alarm1, alarm2, trouble, battery_defect, etc)
But this will generate too much binary_sensor in my opinion, because most of the time the only working attribute is alarm1.
Another option could be create only alarm1, and add an optionnal config parameter for people who wants the full zone status.
Example :
```yaml
zigate:
  full_zone_status: true
```

**Comment by doudz** [comment]
About  information like last detection, or time spent open ,this is already provided by HAss so notre really needed.
Maybe convenient but not mandatory

**Comment by nerdosity** [comment]
I think that we should think about the end user perspective.

- the full_zone_status is a good idea, and it should hide the zigate panel too when not enabled (in my opinion)
- zigate panel (the one with all entities) should probably disabled by default, since it provides too many info
- in the auto-created sensor, only relevant information should be provided. Even if it actually duplicates some info, I think that for battery operated sensor is always relevant battery info, for door sensor the actual status (open or closed) and the time from last event change. I think that the "good" plugin to take inspiration is the default aqara plugin, I actually used it until some day ago since I was deploying my new home alarm installation on a new raspberry, and I only have one zigate device.

So, using experience from previous components (the deconz one was good too, but the app were too "closed" from developers so I sent i back to Amazon... I couldn't get any battery information about three door sensors in one week of usage) we can provide a seemless experience to end user (and to ourselves too), stealing ideas from elsewhere without reinventing the entire wheel.
I alredy started browsing your code, it looks like that the relevant part for supporting new product is in the library, not in this package itself. Am I right?

**Comment by doudz** [comment]
Be aware that I rely on information provided by zigate, and some information sent by device aren't just not recognized by the zigate firmware (I'm not the zigate vendor)
So supporting new device or new message from device could need a firmware upgrade

**Comment by rsokolowski** [comment]
I actually find the full zone information really useful. In particular Xiaomi Water Sensor reports water detection as a change in the zone status (alert1 attribute gets to on). The other HA Zigate component did not support this and I had to change to code on my own to handle zone status (I guess it was to some degree to present nice sensor data instead of bloating the user with all the data).
While I agree that the current behavior is not the best UX one could image (i.e. water and smoke sensor should just be a binary sensor that gets toggle on event detection), I really appreciate that amount of data provided in a sense that I can implement necessary customization on my own.

**Comment by doudz** [comment]
Fix pushed, need to try now

**Comment by nerdosity** [comment]
I will restore the other HA installation with the zigate, will update you ASAP

**Comment by doudz** [comment]
Project is now dead, feel free to fork it

**Comment by nerdosity** [comment]
Hello,

Today I was forced to repair every device after firmware upgrade.
I realized that only one of the SMOK sensor create a zone status sensor, the other one never create a sensor.

![image](https://user-images.githubusercontent.com/26657281/65378593-dbbaf000-dcba-11e9-94a4-346c80c40e4a.png)

I already paired and deleted it three time, a couple of times I was able to remove the sensor with the zigbee command, another time I had to modify the zigate.json.

This create the sensor:



This does not create the sensor:
